### PR TITLE
New .NET SDK for RecoveryServices-DataReplication

### PR DESCRIPTION
**Pull request by minhsuanlee**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by minhsuanlee** [comment]
Skipping "generating samples" and writing scenario tests since this is our very first beta SDK for public preview.

SDK namespace approved as Microsoft.RecoveryServicesDataReplication in https://github.com/Azure/azure-sdk-pr/issues/1220.

Pending for swagger to check-in after ARM's sign-off with https://github.com/Azure/azure-rest-api-specs/pull/24974.

Please go ahead with the review of this PR. It's currently using a swagger PR generated link to swagger and will be replaced and regenerated with official swagger link once swagger PR merged.

File: sdk/applicationinsights/ci.yml
Comment (ArthurMa1978) [inline sdk/applicationinsights/ci.yml:5]: revert this

File: sdk/operationalinsights/ci.yml
Comment (ArthurMa1978) [inline sdk/operationalinsights/ci.yml:6]: revert this

File: sdk/recoveryservices-datareplication/Azure.ResourceManager.RecoveryServicesDataReplication/CHANGELOG.md
Comment (ArthurMa1978) [inline sdk/recoveryservices-datareplication/Azure.ResourceManager.RecoveryServicesDataReplication/CHANGELOG.md:3]: Please specify the release date here by replacing the `Unreleased`, the format is `yyyy-mm-dd`

File: sdk/recoveryservices-datareplication/Azure.ResourceManager.RecoveryServicesDataReplication/README.md
Comment (ArthurMa1978) [inline sdk/recoveryservices-datareplication/Azure.ResourceManager.RecoveryServicesDataReplication/README.md:3]: Describe your service briefly

File: sdk/recoveryservices-datareplication/Azure.ResourceManager.RecoveryServicesDataReplication/src/autorest.md
Comment (ArthurMa1978) [inline sdk/recoveryservices-datareplication/Azure.ResourceManager.RecoveryServicesDataReplication/src/autorest.md:10]: point to the public repo instead of local file path

File: sdk/recoveryservices-datareplication/Azure.ResourceManager.RecoveryServicesDataReplication/src/Properties/AssemblyInfo.cs
Comment (ArthurMa1978) [inline sdk/recoveryservices-datareplication/Azure.ResourceManager.RecoveryServicesDataReplication/src/Properties/AssemblyInfo.cs:6]: According to the swagger pr, the value here is `Microsoft.DataReplication`.

Comment (minhsuanlee) [reply sdk/recoveryservices-datareplication/Azure.ResourceManager.RecoveryServicesDataReplication/src/Properties/AssemblyInfo.cs:6]: Our approved RP namespace is "Microsoft.RecoveryServicesDataReplication" from https://github.com/Azure/azure-sdk-pr/issues/1220. I will update swagger pr folder name to reflect this,

File: sdk/recoveryservices-datareplication/Azure.ResourceManager.RecoveryServicesDataReplication/CHANGELOG.md
Comment (minhsuanlee) [reply sdk/recoveryservices-datareplication/Azure.ResourceManager.RecoveryServicesDataReplication/CHANGELOG.md]: Updated to be 2023-09-30

File: sdk/recoveryservices-datareplication/Azure.ResourceManager.RecoveryServicesDataReplication/src/autorest.md
Comment (minhsuanlee) [reply sdk/recoveryservices-datareplication/Azure.ResourceManager.RecoveryServicesDataReplication/src/autorest.md]: Started using https://github.com/Azure/azure-rest-api-specs/blob/64cddc58b25ae7d6e50cc0dfb5bc19af4ea23f65/specification/recoveryservicesdatareplication/resource-manager/Microsoft.DataReplication/preview/2021-02-16-preview/recoveryservicesdatareplication.json

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**
